cmd/main.go

```diff
@@ -19,9 +19,12 @@ package main
 import (
 	"log"
```

Comment:
Following what still missing be addresseed from the review: 

- a) Could you please supplement the comment to clarify what is expected get done in the todo here? What is the acceptance criteria to remove this todo? ; https://github.com/kubernetes-sigs/kubebuilder/pull/2338/files#r737212666
- b) Remove the commented code in: https://github.com/kubernetes-sigs/kubebuilder/pull/2338/files#r771596382
- c) About changing the API to allow the plugin version (APIVersion) and the plugin Help to be empty, see: https://github.com/kubernetes-sigs/kubebuilder/pull/2338/files#r737215035 and https://github.com/kubernetes-sigs/kubebuilder/pull/2338/files#r737215379. Can we revert that and ensure that any plugin will still have the version and help? 
- d) Should we not use the API version of the plugin? (https://github.com/kubernetes-sigs/kubebuilder/pull/2338/files#r737217533) Why we are seeing it fixed? Can we change it?
- e) Address Nit: https://github.com/kubernetes-sigs/kubebuilder/pull/2338/files#r771598265
- f) Address the request in: https://github.com/kubernetes-sigs/kubebuilder/pull/2338/files#r771603022

However, for we completed Plugin phase 2 still missing:

- a) docs; we need to update the [Plugin Section](https://book.kubebuilder.io/plugins/plugins.html)  
- b) we need to provide an example external plugin for users (it could be under the testdata dir) and could be a plugin that just scaffolds a test file or output hello world
- c) we need to add an e2e test calling the sample plugin and check its output. 

Btw, how can I test your PR to ensure that it works? 
How can we verify it? 

Could you please address these ones? Also, feel free to reach out if you need help.